src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -90,20 +90,23 @@ protected struct Flags
             private const int IsVarargBit = 1 << IsVarargOffset;
 
             private const int ReturnsVoidBit = 1 << ReturnsVoidOffset;
-            private const int ReturnsVoidIsSetBit = 1 << ReturnsVoidOffset + 1;
 
             private const int IsNullableAnalysisEnabledBit = 1 << IsNullableAnalysisEnabledOffset;
 
-            public bool TryGetReturnsVoid(out bool value)
-            {
-                int bits = _flags;
-                value = (bits & ReturnsVoidBit) != 0;
-                return (bits & ReturnsVoidIsSetBit) != 0;
-            }
+            public bool ReturnsVoid
+                => (_flags & ReturnsVoidBit) != 0;
 
             public void SetReturnsVoid(bool value)
             {
-                ThreadSafeFlagOperations.Set(ref _flags, (int)(ReturnsVoidIsSetBit | (value ? ReturnsVoidBit : 0)));
+                if (value)
+                {
+                    ThreadSafeFlagOperations.Set(ref _flags, ReturnsVoidBit);
+                }
+                else
+                {
+                    // We never have to unset this flag
+                    Debug.Assert(!ReturnsVoid);
```

Comment:
Don't understand this assert - in theory, we could allow `SetReturnsVoid(true); SetReturnsVoid(false)`, right? That's different from removing `ReturnsVoidIsSetBit` which wasn't technically possible before.

Reply:
There is no scenario where we set `true` initially but later reset to `false`. The old code would not have been correct for such transition, which is just never exercised. 

The choices in my mind were:
1. just an assertion
2. conditionally throw "unreachable"
3. assert and Unset
4. just Unset

Going with option 4, a reader would wonder why/when it could happen. I'd be okay with either of the remaining options, but I think the assertion (option 1) is simple and clarifies/codifies the existing initialization pattern.